src/science/canopy/cbl_latent_heat.F90

```diff
@@ -82,34 +192,50 @@ SUBROUTINE Latent_heat_flux( mp, CTFRZ, dels, soil_zse, soil_swilt,           &
 
      canopy_fess(j) = MIN(canopy_fess(j), REAL(fupper_limit(j),r_2))
 
-     !Ticket 137 - case iii)
-     !evaporation from frozen soils needs to respect the assumption that
-     !ice fraction of soil moisture cannot exceed frozen_limit=0.85
-     !see soilsnow: if frozen_limit changes need to be consistent
+!|     - Limit 2: Additionally for case 3, the evaporation of liquid water
+!        from within the frozen soil column must not reduce the liquid water fraction
+!        to the point that that ice fraction of soil moisture exceeds an upper limit
+!        frozen_limit=0.85.  This provides a second upper limit on the evaporation and
+!        soil latent flux. **WARNING** frozen_limit=0.85 is hard coded - if it is changed
+!        then the corresponding limit in [[cbl_soilsnow]] must also be changed.
+!
      fupper_limit(j) = REAL(ssnow_wb(j)-ssnow_wbice(j)/0.85)*frescale(j)
      fupper_limit(j) = MAX(REAL(fupper_limit(j),r_2),0.)
 
      canopy_fess(j) = MIN(canopy_fess(j), REAL(fupper_limit(j),r_2))
 
-  END IF
-
+  END IF   
+!*     - The case of dew fall onto a surface with little/no snow while
+!        the soil surface temperature is above freezing is unrestricted and
+!        the soil latent heat flux defaults to the first estimate.
+  
   ssnow_cls(j)=1.
 
-  !Ticket 137 - case ii) deposition of frost onto snow
-  ! case of sublimation of snow overwrites later
+!|     - If the surface has snow cover, or `ssnow_potev`<0 **and** the soil
+!        temperature is below freezing (frost), then the latent heat flux represents a
+!        conversion between solid and vapour phases of water, \(c_{ls}\)=1.1335.
+!        The first estimate for the soil latent heat flux is updated by the change
+!        in value of \(c_{ls}\).
+!
   IF (ssnow_snowd(j) >=0.1 ) THEN
      ssnow_cls(j) = 1.1335
      canopy_fess(j) = ssnow_cls(j)*ssnow_potev(j)
   ENDIF
 
-  !Ticket 137 - case iv) deposition of frost onto frozen soil, no snow
+!|     - For case 4, deposition of frost onto frozen surface (temperature below
+!        freezing) occurs even if there is no snow - there is no limit
+!        on the magnitude of the soil latent heat flux.
+!
   IF (ssnow_snowd(j) < 0.1 .AND. ssnow_potev(j) < 0. .AND. &
        ssnow_tss(j)<CTFRZ) THEN
      ssnow_cls(j)=1.1335
      canopy_fess(j) = ssnow_cls(j)*ssnow_potev(j)
```

Comment:
This has nothing to do with the documentation. Why isn't "ssnow_wetfac" used to calculate canopy_fess?

Reply:
see above - the slight difference here is that the snow pack/soil doesn't play any role in controlling a flux of water when the flux is from the atmosphere onto the soil (it's the atmosphere water content that controls things).  In this set up ssnow_wetfac does get set to 1 earlier on in the SUBROUTINE.